core/domain/collection_services.py

```diff
@@ -491,6 +491,36 @@ def get_collection_summaries_matching_query(query_string, cursor=None):
     ], search_cursor)
 
 
+def get_displayable_collections_matching_query(query_string,
```

Comment:
That's a great question that I don't have an answer to. It seems like ``get_collection_summaries_matching_query`` could easily be changed to ``get_collection_ids_matching_query`` for better parity with explorations. Feel free to make the change. ``collection_services`` also has a ``get_collection_summaries_matching_ids`` function which should come in handy here, I'd think. Please also update corresponding documentation in this method (after it's moved) and the renamed ``get_collection_ids_matching_query`` function after it's renamed. I think that function also has tests that might need to be adjusted.

Thanks, @amgowano!